### ISSUE TITLE
html-webpack-plugin-before-html-processing event not triggered

### ISSUE THREAD

**Issue by JeffreyArts** [ampedandwired/html-webpack-plugin#564]
Hi,

When should the `html-webpack-plugin-before-html-processing` event be triggered? Since I am debugging an (other) plugin, and the code within the function bound to this event is not being triggered.
```
compiler.plugin('compilation', (compilation) => {
    compilation.plugin(
        'html-webpack-plugin-before-html-processing',
        (htmlPluginData, callback) => {
            (...)
        }
    )
});
```
The code within the 'compilation' section is being executed, but the code within the `html-webpack-plugin-before-html-processing` function is not. So in order to find out why, I need to know when this event is triggered, and which options might influence the behavior of this event. Since this should probably give me an explanation of why my ejs variables are not being compiled neither.

**Comment by jantimon** [comment]
The event is fired here:

https://github.com/ampedandwired/html-webpack-plugin/blob/7a781a4120158dcc7329489f280e190960fca81f/index.js#L122

It is only fired if the result is not cached already (you could also try to deactivate the cache).

Please let me know if this helped

**Comment by JeffreyArts** [comment]
Thank you for your reply. Found the solution cause of it. Brief explanation of what happened;

I have the following in my package.json "html-webpack-plugin": "^1.7.0" of the project I am working all. The package.json of the installed package (html-webpack-plugin) shows "version": "1.7.0".

On NPM I noticed that the latest version of this package is *2.26.0*, so I tried to install this version directly by running npm i html-webpack-plugin@2.26.0, and that resolved the problem
